operators/validate_nid.go

```diff
@@ -61,13 +60,14 @@ func (o *validateNid) Evaluate(tx *coraza.Transaction, value string) bool {
 	return res
 }
 
+var nonDigitOrK = regexp.MustCompile(`[^\dk]`)
+
 func nidCl(nid string) bool {
 	if len(nid) < 8 {
 		return false
 	}
-	re := regexp.MustCompile(`[^\dk]`)
 	nid = strings.ToLower(nid)
-	nid = re.ReplaceAllString(nid, "")
+	nid = nonDigitOrK.ReplaceAllString(nid, "")
 	rut, _ := strconv.Atoi(nid[:len(nid)-1])
```

Comment:
Not an issue with this PR, but it looks like a string with a `k` in the middle will be preserved and entered here. If `k` is only meant to be at the end of the string, then that could be checked explicitly instead. Doesn't have to block this PR if we don't know though

Reply:
According to the tests, `k` or `K` only shows up at the end. I am not 100% sure what `@validateNid` is used for, a quick research tells me it is an ID number but I can not confirm 100%. Maybe @jptosso or @piyushroshan can tell? We can take it in a next PR.